File: GXpert/GXpert.Web/Modules/ServerTypes/Skills/SkillCategoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface SkillCategoryRow {
    Id?: number;
    Name?: string;
    Description?: string;
    IsActive?: boolean;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class SkillCategoryRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Skills.SkillCategory';
    static readonly lookupKey = 'Skills.SkillCategory';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<SkillCategoryRow>('Skills.SkillCategory') }
    static async getLookupAsync() { return getLookupAsync<SkillCategoryRow>('Skills.SkillCategory') }

    static readonly deletePermission = 'Skills:SkillsManagement:Modify';
    static readonly insertPermission = 'Skills:SkillsManagement:Modify';
    static readonly readPermission = 'Skills:SkillsManagement:View';
    static readonly updatePermission = 'Skills:SkillsManagement:Modify';

    static readonly Fields = fieldsProxy<SkillCategoryRow>();
}

File: GXpert/GXpert.Web/Modules/Skills/SkillCategory/SkillCategoryRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Serenity.Extensions.Entities;
using System;
using System.ComponentModel;

namespace GXpert.Skills;

[ConnectionKey("Default"), Module("Skills"), TableName("SkillCategory")]
[DisplayName("Skill Category"), InstanceName("Skill Category")]
[ReadPermission(PermissionKeys.SkillsManagement.View)]
[ModifyPermission(PermissionKeys.SkillsManagement.Modify)]
[ServiceLookupPermission("Administration:General")]
[LookupScript("Skills.SkillCategory")]
public sealed class SkillCategoryRow : LoggingRow<SkillCategoryRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Name"), Size(500), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("Description"), Size(500)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Is Active"), NotNull,DefaultValue(1)]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public StringField Name;
        public StringField Description;
        
        public BooleanField IsActive;

    }
}